lib/Conversion/SeqToSV/SeqToSV.cpp

```diff
@@ -190,6 +190,35 @@ class ClockMuxLowering : public OpConversionPattern<ClockMuxOp> {
     return success();
   }
 };
+
+/// Lower `seq.clock_div_2` to a deterministic clock divier
+///
+class ClockDivider2Lowering : public OpConversionPattern<ClockDivider2> {
+public:
+  using OpConversionPattern<ClockDivider2>::OpConversionPattern;
+  using OpConversionPattern<ClockDivider2>::OpAdaptor;
+
+  LogicalResult
+  matchAndRewrite(ClockDivider2 clockDiv, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    Location loc = clockDiv.getLoc();
+
+    Value output = rewriter.create<sv::RegOp>(
+        loc, rewriter.getI1Type(), rewriter.getStringAttr("clock_out"));
+    Value one = rewriter.create<hw::ConstantOp>(loc, APInt(1, 1));
+
+    rewriter.create<sv::AlwaysOp>(
+        loc, sv::EventControl::AtPosEdge, adaptor.getClockIn(), [&] {
+          Value outputVal = rewriter.create<sv::ReadInOutOp>(loc, output);
+          Value inverted = rewriter.create<comb::XorOp>(loc, outputVal, one);
+          rewriter.create<sv::BPAssignOp>(loc, output, inverted);
+        });
+
+    rewriter.replaceOpWithNewOp<sv::ReadInOutOp>(clockDiv, output);
+    return success();
+  }
+};
```

Comment:
I think this makes sense for the behavioral case and is in-line with the clock gate and clock mux lowering above.

But we should also have an option to turn this into an `hw.module.extern` as we do with clock gates, since dedicated ASIC flows (and probably some FPGA flows) will want to provide instantiations of tech-specific stdcells.

Maybe the clock gate externalization pass could be made more general purpose, so it also covers clock dividers? I remember @teqdruid and @mortbopet were starting a discussion on having more general-purpose externalization mechanisms for certain constructs.

Reply:
Another thought: could we make this lower to just a `seq.compreg`?

Reply:
Yeah, we were discussing it in #5719 but are looking for input from others.

Reply:
I agree, but we can add the pass in a follow-up PR.